lockc/src/runc.rs

```diff
@@ -380,7 +373,100 @@ impl RuncWatcher {
             }
         }
 
-        Ok(RuncWatcher { fd })
+        Ok(RuncWatcher {
+            bootstrap_rx,
+            ebpf_tx,
+            fd,
+        })
+    }
+
+    async fn add_container(
+        &self,
+        container_id: String,
+        pid: i32,
+        policy_level: container_policy_level,
+    ) -> Result<(), HandleRuncEventError> {
+        let (responder_tx, responder_rx) = oneshot::channel();
+
+        self.ebpf_tx
+            .send(EbpfCommand::AddContainer {
+                container_id,
+                pid,
+                policy_level,
+                responder_tx,
+            })
+            .await?;
+        responder_rx.await??;
+
+        Ok(())
+    }
+
+    fn add_container_sync(
+        &self,
+        container_id: String,
+        pid: i32,
+        policy_level: container_policy_level,
+    ) -> Result<(), HandleRuncEventError> {
+        debug!("adding container {}", container_id);
+
+        Builder::new_current_thread()
+            .build()?
+            .block_on(self.add_container(container_id, pid, policy_level))?;
+
+        Ok(())
+    }
+
+    async fn delete_container(&self, container_id: String) -> Result<(), HandleRuncEventError> {
+        let (responder_tx, responder_rx) = oneshot::channel();
+
+        self.ebpf_tx
+            .send(EbpfCommand::DeleteContainer {
+                container_id,
+                responder_tx,
+            })
+            .await?;
+        responder_rx.await??;
+
+        Ok(())
+    }
+
+    fn delete_container_sync(&self, container_id: String) -> Result<(), HandleRuncEventError> {
+        debug!("deleting container {}", container_id);
+
+        Builder::new_current_thread()
+            .build()?
+            .block_on(self.delete_container(container_id))?;
+
+        Ok(())
```

Comment:
nitpicking: not needed, use the result from the previous line

Reply:
Same here, I would need to explicitly convert the error from previous line with `map_err`. I prefer staying with `Ok(())`